libs/utils/src/pins.ts

```diff
@@ -2,27 +2,98 @@ import randomBytes from 'randombytes';
 import { EnvironmentFlagName } from './environment_flag';
 import { isFeatureFlagEnabled } from './features';
 
+/** See VVSG 2.0 - 11.3.2-B – Password complexity */
+export const MIN_PIN_LENGTH = 6;
+
+/** Minimum number of unique digits required in a non-weak PIN. */
+const MIN_UNIQUE_DIGITS = 3;
+
+/** Max number of times a digit can repeat, uninterrupted, in a non-weak PIN. */
+const MAX_DIGIT_REPETITION = 3;
+
+/** Max length of sequential string of digits allowed in a non-weak PIN. */
+const MAX_SEQUENTIAL_DIGITS = 3;
+
+/** Long strings of repeated digits that contribute to a weak PIN. */
+const REPEATED_DIGIT_SUBSTRINGS: string[] = ((): string[] => {
+  const repeatedDigitSubstrings: string[] = [];
+
+  for (let digit = 0; digit < 10; digit += 1) {
+    repeatedDigitSubstrings.push(`${digit}`.repeat(MAX_DIGIT_REPETITION + 1));
+  }
+
+  return repeatedDigitSubstrings;
+})();
+
+/** Digit patterns that we consider to contribute to a weak PIN. */
+const SEQUENTIAL_DIGIT_SUBSTRINGS: string[] = ((): string[] => {
+  const sequentialDigitSubstrings: string[] = [];
+
+  for (let digit = MAX_SEQUENTIAL_DIGITS; digit < 10; digit += 1) {
+    const descendingSequence = Array.from({ length: MAX_SEQUENTIAL_DIGITS + 1 })
+      .fill(0)
+      .map((_element, index) => digit - index);
+
+    sequentialDigitSubstrings.push(descendingSequence.join(''));
+    sequentialDigitSubstrings.push(descendingSequence.reverse().join(''));
+  }
+
+  return sequentialDigitSubstrings;
+})();
+
+function newRandomPin(length: number): string {
```

Comment:
> Tangentially: there's probably room for eliminating the option of specifying PIN length altogether, unless we plan on having PINs of different lengths used in different parts of the system at some point

Agreed! I think it's unlikely that we'll allow users to pick a PIN length other than 6, and I like the idea of simplifying